src/commands/fun/xkcd.ts

```diff
@@ -66,7 +66,7 @@ export default class extends Command {
 		interaction.fetchReply().then(reply => { replyId = reply.id; });
 
 		const collector = interaction.channel.createMessageComponentCollector({
-			filter: i => i.message.id === replyId
+			filter: i => i.message.id === replyId && i.user.id === interaction.user.id
```

Comment:
What does this actually do if the interaction isn't collected? Is it still replied to?

Reply:
For the user who pressed the button, if they didn't create the interaction they'd get an "interaction failed", but the collector will still wait until the interaction creator presses a button, at that point it will respond as normal.

Reply:
I'd rather we filter out non-creator button pressers with an ephemeral "hey, you didn't run this command so you cant mess with it" than an "interaction failed"

Reply:
I've tried to do that multiple ways and each of them have been problematic one way or another.

Referring to the following code snippet in rockpaperscissors.ts, for example:
```
collector.on('collect', async (i: ButtonInteraction) => {
  if (interaction.user.id !== i.user.id) {
    await i.reply({
	    content: 'You cannot respond to a command you did not execute',
	    ephemeral: true
    });
  }
}
```
On the third line of that snippet:
- Replying to the original interaction with OR without a return just results in a "this interaction failed" and also stops the command from executing
- Replying to the button interaction with a return does respond to that user, but stops the command execution entirely meaning the original user cannot respond to it
- Replying to the button interaction without a return does respond but falls through the command as if the button presser was the original command executor

Reply:
Actually please disregard all of the above, removing the `max` part of the filter has fixed it for me.